src/Marker.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  Marker.swift
+//  ios-sdk
+//
+//  Created by Sarah Lensing on 3/22/17.
+//  Copyright © 2017 Mapzen. All rights reserved.
+//
+
+import Foundation
+import TangramMap
+
+public enum MarkerType : Int {
+  case currentLocation, searchPin, routeLine
+}
+
+@objc(MZMarker)
+public class Marker: NSObject {
+
+  private static let kPointStyle = "points"
+  private static let kLineStyle = "lines"
+  private static let kPolygonStyle = "polygons"
+  private static let kDefaultBackgroundColor = UIColor.white
+  private static let kDefaultInteractive = true
+  private static let kDefaultSize = CGSize.zero
+
+  public let tgMarker: TGMarker
+  private var styleType = kPointStyle
+  private var markerType: MarkerType?
+  // all marker types have an associated styling path
+  private static let typeToStylingPath = [MarkerType.currentLocation : "layers.mz_current_location_gem.draw.ux-location-gem-overlay",
+                                   MarkerType.searchPin : "layers.mz_search_result.draw.ux-icons-overlay",
+                                   MarkerType.routeLine : "layers.mz_route_line.draw.ux-route-line-overlay"]
+  //currently only search results have an inactive state
+  private static let typeToInactiveStylingPath = [MarkerType.searchPin : "layers.mz_search_result.inactive.draw.ux-icons-overlay"]
+
+  open var point: TGGeoPoint? {
+    set {
+      guard let p = newValue else { return }
```

Comment:
I see the pattern here, we don't provide direct way to remove the geometry from it, only replacement. You can adjust the visibility of it to make it not visible but still usable in the current map, or remove the marker from the map view and then use it again later in the same or a different view.